lib/braintreehttp/serializer/multipart.js

```diff
@@ -40,22 +43,38 @@ class Multipart {
     return /^multipart\/.*/;
   }
 
-  _filePart(key, readStream, boundary) {
+  _filePart(key, readStream, boundary, request) {
     return Buffer.concat([
-      Buffer.from(this._partHeader(key, path.basename(readStream.path), boundary)),
+      Buffer.from(this._partHeader(key, path.basename(readStream.path), boundary, request)),
       fs.readFileSync(readStream.path),
       Buffer.from(Multipart._CRLF)
     ]);
   }
 
-  _formPart(key, formPart, boundary) {
+  _formPart(key, formPart, boundary, request) {
+    let formPartContentType = null;
+    let formPartValue = null;
+    let contentBuffer = null;
+
+    if (formPart instanceof FormPart) {
+      formPartContentType = formPart.headers['Content-Type'];
+      formPartValue = formPart.value;
+    }
+
+    if (formPartContentType === 'application/json') {
+      contentBuffer = Buffer.from(JSON.stringify(formPartValue) + Multipart._CRLF)
```

Comment:
I think this is fine for now, but we should be prepared to support other content types in the future.

Reply:
It's a bit of a hack, but you could construct a new encoder here and pass the part directly in. The encoder will handle serializing the data since there's already a `header` property with `Content-Type` set.

Reply:
We had to break a cyclical dependency in the `Reuse Encoder in Multipart`. Let us know what you think. It does allow us to break the dependency between Encoder and serializers as well to allow specific serializers to be used per Encoder.